src/TrafficLightSimulator.java

```diff
@@ -0,0 +1,76 @@
+
+
+public class TrafficLightSimulator implements Runnable {
+
+  private Thread thrd; // holds the thread that runs the simulation 
+  private TrafficLightColor tlc; // holds the current traffic light color 
+  boolean stop = false; // set to true to stop the simulation 
+  boolean changed = false; // true when the light has changed
+ 
+  TrafficLightSimulator(TrafficLightColor init) {  
+    tlc = init; 
+ 
+    thrd = new Thread(this); 
+    thrd.start(); 
+  } 
+ 
+  TrafficLightSimulator() {  
+    tlc = TrafficLightColor.RED; 
+ 
+    thrd = new Thread(this); 
+    thrd.start(); 
+  } 
+ 
+  // Start up the light. 
+  public void run() { 
+    while(!stop) { 
+      try { 
+      	// No need for a switch here
+    		Thread.sleep( tlc.getDelay() );
+      } catch(InterruptedException exc) { 
+        System.out.println(exc); 
+      } 
+      changeColor(); 
+    }  
+  } 
+ 
+  // Change color. 
+  synchronized void changeColor() { 
+    switch(tlc) { 
+      case RED: 
+        tlc = TrafficLightColor.GREEN; 
+        break; 
+      case YELLOW: 
+        tlc = TrafficLightColor.RED; 
+        break; 
+      case GREEN: 
+       tlc = TrafficLightColor.YELLOW; 
+    } 
```

Comment:
Any thoughts on how to change the colour using the ordinal values? It was optional in the question but it seems most other people have answered it.